lib/chainer/functions/connection/embed_id.rb

```diff
@@ -10,36 +10,38 @@ def self.embed_id(x, w, ignore_label: nil)
           self.new(ignore_label: ignore_label).(x, w)
         end
 
-        def forward_cpu(inputs)
+        def forward(inputs)
+          xm = Chainer.get_array_module(*inputs)
           (x, w) = inputs
 
           unless @ignore_label
-            return [Chainer::Utils::Array.take(w, x.to_a, axis: 0)]
+            return [Chainer::Utils::Array.take(w, x, axis: 0)]
           end
 
-          mask = Numo::Bit.cast(x.flatten.to_a.map{|i| i == @ignore_label}).reshape(*x.shape)
+          mask = xm::Bit.cast(x.flatten.to_a.map{|i| i == @ignore_label}).reshape(*x.shape)
           if mask.where.empty?
```

Comment:
I did not measure, but `mask.count == 0` is faster maybe?

Reply:
I was surprised. Surely It is doing the same movement.

Reply:
I did not try, but might be “count_true”, then.